networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/AttestationTopicSubscriber.java

```diff
@@ -36,25 +40,62 @@ public AttestationTopicSubscriber(final Eth2Network eth2Network) {
   public synchronized void subscribeToCommitteeForAggregation(
       final int committeeIndex, final UnsignedLong aggregationSlot) {
     final int subnetId = committeeIndexToSubnetId(committeeIndex);
-    final UnsignedLong currentUnsubscriptionSlot =
-        unsubscriptionSlotBySubnetId.getOrDefault(subnetId, ZERO);
+    final UnsignedLong currentUnsubscriptionSlot = allSubscriptions.getOrDefault(subnetId, ZERO);
     if (currentUnsubscriptionSlot.equals(ZERO)) {
       eth2Network.subscribeToAttestationSubnetId(subnetId);
     }
-    unsubscriptionSlotBySubnetId.put(subnetId, max(currentUnsubscriptionSlot, aggregationSlot));
+    allSubscriptions.put(subnetId, max(currentUnsubscriptionSlot, aggregationSlot));
+  }
+
+  public synchronized void subscribeToPersistentSubnets(
+      final Set<SubnetSubscription> newSubscriptions) {
+    boolean shouldUpdateENR = false;
+
+    for (SubnetSubscription subnetSubscription : newSubscriptions) {
+      int subnetId = subnetSubscription.getSubnetId();
+      if (!persistentSubnetIdSet.contains(subnetId)) {
+        persistentSubnetIdSet.add(subnetId);
+        shouldUpdateENR = true;
+      }
```

Comment:
(optional) How about:
```suggestion
    shouldUpdateENR = subnetIds.stream().map(persistentSubnetIdSet::add).reduce((a,b) -> a || b)
```
😄

Reply:
that's pretty cool